openfe/tests/protocols/test_openmm_equil_rfe_protocols.py

```diff
@@ -458,7 +461,58 @@ def test_dry_run_ligand_tip4p(benzene_system, toluene_system,
     with tmpdir.as_cwd():
         sampler = dag_unit.run(dry=True)['debug']['sampler']
         assert isinstance(sampler, MultiStateSampler)
-        assert sampler._factory.hybrid_system
+
+        # Get the htf and check
+        # 1. The number of particles in the system are as expected
+        #   - tot particles = old particles + new unique
+        # 2. The number of vs == number of waters
+        # 3. The vs parameters are as expected
+        htf = sampler._factory
+
+        # Test 1
+        old_particle_count = htf._old_system.getNumParticles()
+        unique_new_count = len(htf._unique_new_atoms)
+        hybrid_particle_count = htf.hybrid_system.getNumParticles()
+        assert old_particle_count + unique_new_count == hybrid_particle_count
+
+        # Test 2
+        num_waters = len(
+            [r for r in htf._old_topology.residues() if r.name =='HOH']
+        )
+        virtual_sites = [
+            ix for ix in range(htf.hybrid_system.getNumParticles()) if
+            htf.hybrid_system.isVirtualSite(ix)
+        ]
+        assert num_waters == len(virtual_sites)
+
+        # Test 3
+        # get the standard and custom nonbonded forces - one of each
+        nonbond = [f for f in htf.hybrid_system.getForces()
+                   if isinstance(f, NonbondedForce)][0]
+
+        cust_nonbond = [f for f in htf.hybrid_system.getForces()
+                        if isinstance(f, CustomNonbondedForce)][0]
+
+        # loop through every virtual site and check that they have the
+        # expected tip4p parameters
+        for entry in virtual_sites:
+            vs = htf.hybrid_system.getVirtualSite(entry)
+            vs_mass = htf.hybrid_system.getParticleMass(entry)
+            assert ensure_quantity(vs_mass, 'openff').m == pytest.approx(0)
+            vs_weights = [vs.getWeight(ix) for ix in range(vs.getNumParticles())]
+            np.testing.assert_allclose(
+                vs_weights, [0.786646558, 0.106676721, 0.106676721]
+            )
+            c, s, e = nonbond.getParticleParameters(entry)
+            assert ensure_quantity(c, 'openff').m == pytest.approx(-1.04844)
```

Comment:
should we look at contributing some sort of `assert_quantity_equal(val, expected_mag, expected_unit)` to openff units?

Reply:
If Matt wants to? We could also just make our own tooling if it's necessary.